apps/site/src/lib/components/Navbar/Navbar.tsx

```diff
@@ -0,0 +1,87 @@
+"use client";
+
+import * as NavMenu from "@radix-ui/react-navigation-menu";
+
+import styles from "./Navbar.module.scss";
+import React from "react";
+import { useState, useEffect } from "react";
+
+import NavLinkItem from "./NavbarHelpers";
+import Button from "@/lib/components/Button/Button";
+import HackLogo from "@/lib/components/HackLogo/HackLogo";
+
+import hamburger from "@/assets/icons/navigation-icon.svg";
+import Image from "next/image";
+
+function Navbar() {
+	const [listShown, setListShown] = useState(false);
+	const [transitionApplied, setTransitionApplied] = useState(false);
+	const [hasScrolled, setHasScrolled] = useState(false);
+
+	useEffect(() => {
+		const mobileNavInitStateHandler = () => {
+			setListShown(false);
+			setTransitionApplied(false);
+		};
+
+		// hides list on breakpoint
+		window
+			.matchMedia("(min-width: 768px)")
+			.addEventListener("change", mobileNavInitStateHandler);
+	}, []);
```

Comment:
Thought: this seems more complicated than it needs to be. There should be a way to hide the list with CSS.

Reply:
When the window gets smaller and hits the md breakpoint, the transition is applied and animates the list hiding, which isn't supposed to happen. I think that's why I added that event listener so that on the md breakpoint, the list doesn't animate and only does so when the button is pressed again. I don't exactly know how to fix this other than the event listener for a window size change.